v2/tools/generator/internal/astmodel/local_package_reference.go

```diff
@@ -138,6 +138,55 @@ func (pr LocalPackageReference) GroupVersion() (string, string) {
 	return pr.group, pr.Version()
 }
 
+// ImportAlias returns the import alias to use for this package reference
+func (pr LocalPackageReference) ImportAlias(style PackageImportStyle) string {
+	switch style {
+	case VersionOnly:
+		return fmt.Sprintf(
+			"%s%s",
+			pr.simplifiedGeneratorVersion(pr.generatorVersion),
+			pr.simplifiedApiVersion(pr.apiVersion))
+	case GroupOnly:
+		return pr.group
+	case GroupAndVersion:
+		return fmt.Sprintf(
+			"%s_%s%s",
+			pr.group,
+			pr.simplifiedGeneratorVersion(pr.generatorVersion),
+			pr.simplifiedApiVersion(pr.apiVersion))
+	default:
+		panic(fmt.Sprintf("didn't expect PackageImportStyle %q", style))
+	}
+}
+
+func (pr LocalPackageReference) simplifiedApiVersion(version string) string {
+	return strings.ToLower(pr.simplify(version, apiVersionSimplifications))
+}
+
+var apiVersionSimplifications = map[string]string{
+	"alpha":   "a",
+	"beta":    "b",
+	"preview": "p",
+	"-":       "",
+}
+
+func (pr LocalPackageReference) simplifiedGeneratorVersion(version string) string {
+	return pr.simplify(version, generatorVersionSimplifications)
+}
+
+var generatorVersionSimplifications = map[string]string{
+	"v1alpha1api": "alpha",
+	"v1beta1api":  "beta",
+	"v1api":       "v",
+}
+
+func (pr LocalPackageReference) simplify(result string, simplifications map[string]string) string {
+	for l, s := range simplifications {
+		result = strings.Replace(result, l, s, -1)
```

Comment:
Consider using a generic `strings.NewReplacer` here for both `apiVersionSimplifications` and `generatorVersionSimplifications`

Reply:
That's really nice, thanks for the pointer. Changed.